src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs

```diff
@@ -196,7 +196,7 @@ XmlNode ITestRunner.Run(ITestEventListener listener, TestFilter filter)
         ITestRun ITestRunner.RunAsync(ITestEventListener listener, TestFilter filter)
         {
             var testRun = new TestRun(this);
-            testRun.Start(listener, filter);
+            testRun.RunAsync(listener, filter);
             return testRun;
```

Comment:
Well, it was void because it didn't return anything. :-)

Combining TestRun and AsyncTestEngineResult sounds good to me, provided you clearly keep the interfaces separate. ITestRun is intended for use by clients and shouldn't have any reference to the internal TestEngineResult in it.

BTW: You probably understood this already, but just in case, TestEngineResult exists for the sole purpose of avoiding unnecessary parsing  and construction of XmlNodes as results are transmitted between different processes and machines. The user only deals with a __single__ XmlNode that combines all results.